np.sh

```diff
@@ -1,8 +1,23 @@
 #!/usr/bin/env sh
+
+if test -n "$(git status -uno --porcelain)"; then
+	echo "Unclean working tree. Commit or stash changes first." >&2;
+	exit 128;
+fi
+
+if ! git fetch --quiet 2>/dev/null; then
+    echo "There was a problem fetching your branch" >&2;
+    exit 128;
+fi
+
+if test "00" != $(git rev-list --count --left-right @'{u}'...HEAD | awk '{ print $1$2 }'); then
```

Comment:
Btw, the quotes around the sub-shell expression are important here, since if the command doesn't produce the expected output, you might run into something like this:

```sh
sh-3.2$ test "00" != $(git rev-list --count --left-right @'{u}'...HEAD | awk '{print $1$2}')
fatal: Not a git repository (or any of the parent directories): .git
sh: test: 00: unary operator expected
```